diff --git a/config.mak b/config.mak
index 1911e89..161f37d 100644
--- a/config.mak
+++ b/config.mak
@@ -346,20 +346,7 @@ core/config.h: config.inc core/version.h tools/config.sh config.mak
 	@${MAKE} -s -f config.mak config.h.echo >> core/config.h

 # TODO: fix when git is not available
-core/version.h: $(shell git show-ref HEAD | ${SED} "s,^.* ,.git/,g")
-	@${ECHO} MAKE $@
-	@${ECHO} "/* created by ${MAKE} -f config.mak */" > core/version.h
-	@${ECHO} "#define POTION_MAJOR ${POTION_MAJOR}" >> core/version.h
-	@${ECHO} "#define POTION_MINOR ${POTION_MINOR}" >> core/version.h
-	@${ECHO} "#define POTION_VERSION \"${POTION_MAJOR}.${POTION_MINOR}\"" >> core/version.h
-	@${ECHO} -n "#define POTION_DATE   \"" >> core/version.h
-	@${ECHO} -n $(shell date +%Y-%m-%d) >> core/version.h
-	@${ECHO} "\"" >> core/version.h
-	@${ECHO} -n "#define POTION_COMMIT \"" >> core/version.h
-	@${ECHO} -n $(shell git rev-list HEAD -1 --abbrev=7 --abbrev-commit) >> core/version.h
-	@${ECHO} "\"" >> core/version.h
-	@${ECHO} -n "#define POTION_REV    " >> core/version.h
-	@${ECHO} -n $(shell git rev-list --abbrev-commit HEAD | wc -l | ${SED} "s/ //g") >> core/version.h
-	@${ECHO} >> core/version.h
+core/version.h:
+	@echo static file

-.PHONY: config config.inc.echo config.h.echo
+.PHONY: config config.inc.echo config.h.echo core/version.h
